pkg/collector/corechecks/cluster/topologycollectors/service_collector.go

```diff
@@ -212,6 +227,65 @@ func (sc *ServiceCollector) serviceToStackStateComponent(service v1.Service) *to
 	return component
 }
 
+func (sc *ServiceCollector) serviceToExternalServiceComponent(service v1.Service) *topology.Component {
+	log.Tracef("Mapping kubernetes pod ExternalName service to extra StackState component: %s", service.String())
+	// create identifier list to merge with StackState components
+	identifiers := make([]string, 0)
+
+	if service.Spec.ExternalName != "None" && service.Spec.ExternalName != "" {
+		identifiers = append(identifiers, fmt.Sprintf("urn:endpoint:/%s", service.Spec.ExternalName))
+		// If targetPorts are specified, use those
+		for _, port := range service.Spec.Ports {
+			// map all the node ports
+			if port.Port != 0 {
+				identifiers = append(identifiers, fmt.Sprintf("urn:endpoint:/%s:%s:%d", sc.GetInstance().URL, service.Spec.ExternalName, port.Port))
+			}
+		}
+
+		addrs, err := sc.DNS(service.Spec.ExternalName)
+		if err != nil {
+			log.Warnf("Could not lookup IP addresses for host '%s' (Error: %s)", service.Spec.ExternalName, err.Error())
+		} else {
+			for _, addr := range addrs {
+				identifiers = append(identifiers, sc.buildEndpointExternalID(addr))
+				// If targetPorts are specified, use those
+				for _, port := range service.Spec.Ports {
+					// map all the node ports
+					if port.Port != 0 {
+						identifiers = append(identifiers, fmt.Sprintf("urn:endpoint:/%s:%s:%d", sc.GetInstance().URL, addr, port.Port))
+					}
+				}
+			}
+		}
+	}
+
+	// add identifier for this service name
+	serviceID := buildServiceID(service.Namespace, service.Name)
+	identifiers = append(identifiers, fmt.Sprintf("urn:external-service:/%s:%s", sc.GetInstance().URL, serviceID))
+
+	log.Tracef("Created identifiers for %s: %v", service.Name, identifiers)
+
+	externalID := fmt.Sprintf("%s:%s:external-service/%s", sc.GetURNBuilder().URNPrefix(), service.Namespace, service.Name)
+
+	tags := sc.initTags(service.ObjectMeta)
```

Comment:
Do you know if service types are part of tags? It might be interesting to be able to filter on those

Reply:
They're not, but I could add it definitely indeed